client/src/pages/reports/Form.js

```diff
@@ -572,6 +575,26 @@ const ReportForm = ({
                       fields={Location.autocompleteQuery}
                       valueKey="name"
                       addon={LOCATIONS_ICON}
+                      createEntityComponent={
+                        (canCreateLocation &&
+                          ((searchText, setDoReset) => (
+                            <LocationForm
+                              initialValues={new Location({ name: searchText })}
+                              title="Create New Location"
+                              afterSaveActions={value => {
+                                // validation will be done by setFieldValue
+                                setFieldTouched("location", true, false) // onBlur doesn't work when selecting an option
+                                setFieldValue("location", value, true)
+                                setDoReset(true)
+                                toast.success("The location has been saved")
+                              }}
+                              afterCancelActions={() => {
+                                setDoReset(true)
+                              }}
+                            />
+                          ))) ||
+                        null
```

Comment:
For readability, I'd suggest to extract this as a component, e.g.:
```javascript
const CreateNewLocation = ({
  canCreateNewLocation,
  searchText,
  setFieldTouched,
  setFieldValue,
  setDoReset
}) => {
  if (!canCreateNewLocation) {
    return null
  }

  return (
    <LocationForm
      initialValues={new Location({ name: searchText })}
      title="Create New Location"
      afterSaveActions={value => {
        // validation will be done by setFieldValue
        setFieldTouched("location", true, false) // onBlur doesn't work when selecting an option
        setFieldValue("location", value, true)
        setDoReset(true)
        toast.success("The location has been saved")
      }}
      afterCancelActions={() => {
        setDoReset(true)
      }}
    />
  )
}

CreateNewLocation.propTypes = {
  canCreateNewLocation: PropTypes.bool,
  searchText: PropTypes.string,
  setFieldTouched: PropTypes.func.isRequired,
  setFieldValue: PropTypes.func.isRequired,
  setDoReset: PropTypes.func.isRequired
}
```
then call it like this:
```javascript
                      createEntityComponent={(searchTerms, setDoReset) => (
                        <CreateNewLocation
                          canCreateNewLocation={canCreateLocation}
                          searchTerms={searchTerms}
                          setFieldTouched={setFieldTouched}
                          setFieldValue={setFieldValue}
                          setDoReset={setDoReset}
                        />
                      )}
```